bokeh/models/legends.py

```diff
@@ -0,0 +1,208 @@
+""" Renderers for various kinds of legends that can be added to
+Bokeh plots
+
+"""
+from __future__ import absolute_import
+
+from ..core.enums import Orientation, LegendLocation
+from ..core.property_mixins import LineProps, FillProps, TextProps
+from ..core.properties import (
+    Int, String, Enum, Instance, List, Dict, Tuple, Include, Either,
+    Float, Override,
+)
+
+from .formatters import TickFormatter, BasicTickFormatter
+from .mappers import ColorMapper
+from .renderers import GuideRenderer, GlyphRenderer
+from .tickers import Ticker, BasicTicker
+
+class Legend(GuideRenderer):
```

Comment:
This change needs an explanation (should have been done in the initial commit). There might be unforeseen consequences of this change across code base. This is also why one should refrain from using "refactoring" near such changes, as this changes both APIs and functionality.